clang-tools-extra/clang-tidy/readability/ImplicitBoolConversionCheck.cpp

```diff
@@ -267,6 +275,10 @@ void ImplicitBoolConversionCheck::registerMatchers(MatchFinder *Finder) {
   auto BoolXor =
       binaryOperator(hasOperatorName("^"), hasLHS(ImplicitCastFromBool),
                      hasRHS(ImplicitCastFromBool));
+  auto ComparisonInCall =
+      allOf(hasParent(callExpr()),
+            has(binaryOperator(hasAnyOperatorName("==", "!="))));
```

Comment:
has -> hasSourceExpression, and probably castExpr would be needed instead of allOf